### PR TITLE
Update User and developer guide

### PR DESCRIPTION
**Pull request by howardwhw2**
User Guide:
- Add page breaks for User Guide
- Add and update images for `viewimage` and `helpwindow`
- Removed empty FAQ section

Developer Guide:
- Updated links in developer guide
- Add TOC for developer guide (`Implementation` portion not done yet)
- Add user stories
- Update value proposition
- Moved glossary

Guide can be viewed [here](https://howardwhw2.github.io/tp/UserGuide.html)

### CODE REVIEW

**Review by waffledood — APPROVED** [review]
LGTM!

File: docs/Documentation.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ title: Documentation guide
 
 **Diagrams:**
 
-* See the [_[se-edu/guides] **Using PlantUML**_](https://se-education.org/guides/tutorials/plantUml.html)
+* Our diagrams are created using [Diagrams.net](https://www.diagrams.net/)
```
Comment (hsiaojietng) [inline docs/Documentation.md:25]: We are using draw.io too right?

File: docs/UserGuide.md [hunk 2]
```diff
@@ -86,6 +90,8 @@ Some example commands you can try:
 
 </div>
 
+<div style="page-break-after: always;"></div>
+
```
Comment (hsiaojietng) [inline docs/UserGuide.md:94]: Thanks a lot for adding the page breaks!

**Review by hsiaojietng — APPROVED** [review]
LGTM just one thing to add

File: docs/Documentation.md (hunk 1 above)
Comment (howardwhw2) [reply docs/Documentation.md:25]: draw.io is diagrams.net

**Review by cindyangXQ — APPROVED** [review]
LGTM

**Pull request merged**